### PR TITLE
dynamic audit documentation

### PR DESCRIPTION
**Pull request by pbarker**
Adds documentation for dynamic audit configuration

### CODE REVIEW

**Comment by Bradamant3** [comment]
n.b. to anyone else who might review this PR: this feature is alpha for 1.12 and there has been discussion in the past about doing things differently for alpha features. Maybe we need to take the issue up in Slack or sig-docs meeting?

**Comment by justaugustus** [comment]
Bump for approval!

**Comment by pbarker** [comment]
this should be moved to the 1.13 milestone now

File: content/en/docs/tasks/debug-application-cluster/audit.md [hunk 1]
```diff
@@ -199,6 +199,44 @@ same format as described above to the aggregated apiserver and set up the log in
...
 
+## Dynamic Auditing
+
+As of version 1.12 audit backends can be configured dynamically as api objects. 
+
+To enable the feature a number of api server flags must be set:
+- `--audit-dynamic-configuration` the primary switch, once the feature progresses to GA this will be 
+  the only flag needed
+- `--feature-gates=DynamicAuditing=true` Feature gate
+- `--runtime-config=audit.k8s.io/v1alpha1=true` Forces alpha version
```
Comment (charrywanganthony) [inline content/en/docs/tasks/debug-application-cluster/audit.md]: should be `admissionregistration.k8s.io/v1alpha1=true`?

**Comment by pbarker** [comment]
@charrywanganthony sorry this was fairly out of date when you looked at it, the feature is now merged and I updated the doc with the changes. Please take another look when you get a chance. cc @Bradamant3

File: content/en/docs/tasks/debug-application-cluster/audit.md [hunk 2]
```diff
@@ -207,6 +207,59 @@ same format as described above to the aggregated apiserver and set up the log in
...
+      url: <backend url>
+      service:
+        name: <service name>
+        namespace: <service namespace>
+      caBundle: <ca bundle>
+```
+
+For the complete API definition, see [the godoc](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/apis/audit/v1alpha1/types.go).
+
+Multiple objects will exist as independent solutions. Dynamic policy will have no effect on backends configured from runtime flags. If webhook truncate options are set as runtime flags they will be applied to the dynamic backend.
```
Comment (Bradamant3) [inline content/en/docs/tasks/debug-application-cluster/audit.md]: This paragraph is a bit confusing:
- Does the first sentence help explain the second and third? (relationship is not quite clear)
- As written, the second and third sentences seem to contradict each other. Dynamic policy has no effect if you configure from runtime flags? But if you set runtime flags they are applied to dynamic policy? As I write these questions I'm also wondering whether I'm confusing "policy" and "backend" -- but I think the paragraph, or possibly the intro content to this section, could be rewritten to make the relationship clearer.

**Comment by Bradamant3** [comment]
Better netlify link: https://deploy-preview-9947--kubernetes-io-vnext-staging.netlify.com/docs/tasks/debug-application-cluster/audit/

**Comment by pbarker** [comment]
@Bradamant3 updated the wording, let me know what you think

**Comment by Bradamant3** [comment]
@pbarker SO MUCH BETTER, THANK YOU! (I did a little light copyediting for style guide compliance, but it was easy this time :-) )

**Comment by Bradamant3** [comment]
@charrywanganthony if you can lgtm, we can get this PR merged. Thanks!

/approve

**Comment by pbarker** [comment]
@Bradamant3 updated the yaml as well, looks like it was rendering wrong with the default fields I had, let me know if that looks good

File: content/en/docs/tasks/debug-application-cluster/audit.md [hunk 3]
```diff
@@ -207,6 +207,55 @@ same format as described above to the aggregated apiserver and set up the log in
 to pick up audit logs. Different apiservers can have different audit configurations and different
 audit policies.
 
+## Dynamic Auditing
+
+{{< feature-state for_k8s_version="v1.13" state="alpha" >}}
+
+In Kubernetes version 1.13, you can configure dynamic audit webhook backends AuditSink API objects. 
+
+To enable dynamic auditing you must set the following apiserver flags:
```
Comment (charrywanganthony) [inline content/en/docs/tasks/debug-application-cluster/audit.md]: --runtime-config=auditregistration.k8s.io/v1alpha1=true

Comment (charrywanganthony) [reply content/en/docs/tasks/debug-application-cluster/audit.md]: it seems to be needed

File: content/en/docs/tasks/debug-application-cluster/audit.md [hunk 4]
```diff
@@ -199,6 +202,56 @@ available for the log backend:
...
+### Dynamic backend
+
+{{< feature-state for_k8s_version="v1.13" state="alpha" >}}
+
+In Kubernetes version 1.13, you can configure dynamic audit webhook backends AuditSink API objects. 
+
+To enable dynamic auditing you must set the following apiserver flags:
+
+- `--audit-dynamic-configuration`: the primary switch. When the feature is at GA, the only required flag.   
+- `--feature-gates=DynamicAuditing=true`: feature gate at alpha and beta.
```
Comment (tengqm) [inline content/en/docs/tasks/debug-application-cluster/audit.md]: Please revise content/en/docs/reference/command-line-tools-reference/feature-gates.md by documenting this feature gate

File: content/en/docs/tasks/debug-application-cluster/audit.md [hunk 5]
```diff
@@ -199,6 +202,56 @@ available for the log backend:
...
+    - ResponseComplete
+  webhook:
+    throttle:
+      qps: 10
+      burst: 15
+    clientConfig:
+      url: "https://audit.app"
+```
+
+For the complete API definition, see [the godoc](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/auditregistration/v1alpha1/types.go). Multiple objects will exist as independent solutions.
```
Comment (tengqm) [inline content/en/docs/tasks/debug-application-cluster/audit.md]: If there is a API resource definition for this, we should avoid linking to source code.

File: content/en/docs/tasks/debug-application-cluster/audit.md [hunk 6]
```diff
@@ -199,6 +202,56 @@ available for the log backend:
...
+
+Existing static backends that you configure with runtime flags are not affected by this feature. However, the dynamic backends share the truncate options of the static webhook. If webhook truncate options are set with runtime flags, they are applied to all dynamic backends.
+
+#### Policy
+
+The AuditSink policy differs from the runtime policy. This is because the API object serves different use cases. The policy will continue to evolve to serve more use cases.
```
Comment (tengqm) [inline content/en/docs/tasks/debug-application-cluster/audit.md]: What do you mean by "runtime policy"?

File: content/en/docs/tasks/debug-application-cluster/audit.md (hunk 4 above)
Comment (charrywanganthony) [reply content/en/docs/tasks/debug-application-cluster/audit.md]: #11292

File: content/en/docs/tasks/debug-application-cluster/audit.md (hunk 6 above)
Comment (pbarker) [reply content/en/docs/tasks/debug-application-cluster/audit.md]: The policy type used in `--audit-policy-file` flag, defined https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/apis/audit/v1/types.go#L153, not sure if there would be a better way of saying that 🤔

File: content/en/docs/tasks/debug-application-cluster/audit.md (hunk 5 above)
Comment (pbarker) [reply content/en/docs/tasks/debug-application-cluster/audit.md]: the swagger is here https://github.com/kubernetes/kubernetes/blob/master/api/swagger-spec/auditregistration.k8s.io_v1alpha1.json or are you referring to the website api documentation? If so I'm not sure hot to properly link to that

File: content/en/docs/tasks/debug-application-cluster/audit.md (hunk 6 above)
Comment (charrywanganthony) [reply content/en/docs/tasks/debug-application-cluster/audit.md]: something like "legacy audit runtime policy"?

**Pull request merged**